### PR TITLE
Support Calculated fields in Collection Object and Preparation

### LINKED ISSUE #1009: Support Calculated fields in Collection Object and Preparation
The latest version of specify 6 added two calculated fields to Collection Object and one to Preparation.  The fields are blank in Specify 7

### PR DESCRIPTION
**Pull request by benanhalt**
fixes #1009

### CODE REVIEW

**Comment by benanhalt** [comment]
The below fields have been added to the API. Now the front-end needs to be updated to support them in forms. The name fields in the forms are as given below after the dot.

collectionobject.totalCountAmt
collectionobject.actualTotalCountAmt

preparation.actualCountAmt

accession.actualTotalCountAmt
accession.totalCountAmt
accession.preparationCount
accession.collectionObjectCount

**Comment by benanhalt** [comment]
This breaks compatibility with Specify 6.8.00.

**Comment by benanhalt** [comment]
This is still needs front end support. Back end has been modified to include the calculated fields in the API responses for the relevant records.

**Comment by maxxxxxdlp** [comment]
@benanhalt Added. Please check if the data types are correct

File: specifyweb/frontend/js_src/lib/schemaextras.ts [hunk 1]
```diff
@@ -227,8 +295,19 @@ export const schemaExtras: {
     isOnLoan.isHidden = true;
     isOnLoan.overrides.isHidden = true;
 
+    const preparation = new LiteralField(model, {
```
Comment (benanhalt) [inline specifyweb/frontend/js_src/lib/schemaextras.ts:298]: Where is this Loan.preparation field coming from? It is not one of the calculated fields that was added?

Comment (maxxxxxdlp) [reply specifyweb/frontend/js_src/lib/schemaextras.ts]: It's not Loan.preparation. It's Preparation.preparation (the diff is collapsed)

Looks like a copy-paste bug. Fixed in [9d52d4f](https://github.com/specify/specify7/commit/9d52d4f041cea0dc6990740a1ed95a6d548f8296)

Comment (benanhalt) [reply specifyweb/frontend/js_src/lib/schemaextras.ts]: Yeah sorry. I just realized that and edited the comment. But there should be a totalCountAmt field.

Comment (benanhalt) [reply specifyweb/frontend/js_src/lib/schemaextras.ts]: The types look correct. All integers.

**Comment by benanhalt** [comment]
Let's not merge this until the next tagged release because it breaks compatibility with 6.8.00 and I don't want to break the edge build for those users.

File: specifyweb/frontend/js_src/lib/schemaextras.ts (hunk 1 above)
Comment (maxxxxxdlp) [reply specifyweb/frontend/js_src/lib/schemaextras.ts]: Fixed in [60bad96](https://github.com/specify/specify7/commit/60bad96ea8d3a0658ea43171f009b3a0d2a69b84)

**Pull request merged**